test/unit/modules/vdom/create-component.spec.js

```diff
@@ -76,6 +76,32 @@ describe('create-component', () => {
     go()
   })
 
+  it('create a component when resolved with synchronous async loading', done => {
+    const data = {
+      props: {},
+      staticAttrs: { class: 'bar' }
+    }
+    spyOn(vm, '$forceUpdate')
+    function async (resolve, reject) {
+      resolve({
+        name: 'child',
+        props: ['msg']
+      })
+    }
+    const vnode = createComponent(async, data, vm, vm)
+    expect(vnode.asyncFactory).toBe(async)
+    expect(vnode.asyncFactory.contexts.length).toEqual(0)
+    expect(vnode.tag).toMatch(/vue-component-[0-9]+-child/)
+    expect(vnode.data.staticAttrs).toEqual({ class: 'bar' })
+    expect(vnode.children).toBeUndefined()
+    expect(vnode.text).toBeUndefined()
+    expect(vnode.elm).toBeUndefined()
+    expect(vnode.ns).toBeUndefined()
+    expect(vnode.context).toEqual(vm)
+     expect(vm.$forceUpdate).not.toHaveBeenCalled()
```

Comment:
Did my suggested patch include an extra space 😅? Sorry for that.

Reply:
Apparently ^^, just fixed it 👍